float32[] angles
---
bool success